torch/csrc/jit/script/compiler.cpp

```diff
@@ -722,9 +737,19 @@ struct to_ir {
         }
       }
       ensureTensors(return_stmt.range(), results);
+      if (typed_def.schema && typed_def.schema->returns.size() != results.size()) {
+        throw ErrorReport(def.range()) << "Number of type annotations for function"
+          << " return (" << typed_def.schema->returns.size() << ") does not match"
+          << " the number of returns from the function (" << results.size() << ")!";
+      }
+      size_t return_type_idx = 0;
       for(auto r : results) {
         graph->registerOutput(r);
-        returns.push_back({"", DynamicType::get()});
+        TypePtr type = DynamicType::get();
+        if (typed_def.schema) {
+          type = typed_def.schema->returns[return_type_idx++].type;
```

Comment:
This is missing a check that the returned type actually matches the schema type. If you wrote something that says returns `int` right now but returns a tensors, that would case this to be wrong.